client/client.go

```diff
@@ -16,7 +17,41 @@ import (
 )
 
 type OpkClient struct {
-	Op OpenIdProvider
+	Op   OpenIdProvider
+	CosP *CosignerProvider
```

Comment:
Marking this as won't fix.

To cut down on interface pollution I avoid creating interfaces until I have more than two examples of a struct. Given that cosigner behavior is strictly defined and we generally don't expect developers to write their own CosignerProvider implementations (we do expect them to write their own Cosigner implementations but not Cosigner Provider). I don't see the benefit of adding an interface here. 

That said, once we get a sense what a generalized behavior an cosigner provider should provide and we see the need for more than on implementation of a cosigner provider, we can decide to replace this with an interface.

Reply:
Yeah I guess I was thinking in terms of unit testing strictly the logic in the `OpkClient` struct, where `CosP` is used, _without_ having to spin up a Cosigner implementation test http server and have the `CosignerProvider` talk to it (which also makes an http server for the callback path) in order to test `OpkClient`; an interface would solve this. 

But that may not be warranted just yet because it looks like `OpkClient` doesn't have too much extra logic besides just calling methods on `Op` and `CosP`.

Side note though: for an integration test of Cosigner logic, we can def do something like that where we spin up a test http server for those endpoints.